plugins/org.eclipse.reddeer.eclipse/src/org/eclipse/reddeer/eclipse/wst/server/ui/wizard/ModifyModulesDialog.java

```diff
@@ -49,9 +49,14 @@ public void finish(TimePeriod timeout) {
 		new FinishButton(this).click();
 		new WaitUntil(new ShellHasChildrenOrIsNotAvailable(getShell()));
 		if(!getShell().isDisposed()){
-			Shell serverShell = new DefaultShell("Server");
-			new OkButton(serverShell).click();
-			new WaitWhile(new ShellIsAvailable(serverShell));
+			try {
+				ShellIsAvailable serverShell = new ShellIsAvailable("Server");
+				new WaitUntil(serverShell);
```

Comment:
how about using WaitUntil(serverShell, TimePeriod.DEFAULT, false)
then if wait's result is not null (meaning shell was found) continue (click ok..)?